internal/topology/variables/patch_validation.go

```diff
@@ -0,0 +1,222 @@
+/*
```

Comment:
Q: is there a reason why patch validation is under internal/topology/variables?
Can this go into internal/webhooks given that it is consumed only there?